src/plugins/aws/ssh.ts

```diff
@@ -33,4 +54,47 @@ export const awsSshProvider: SshProvider<
         };
   },
   toCliRequest: async (request) => ({ ...request, cliLocalData: undefined }),
+  cloudProviderLogin: async (authn, request) => {
+    if (!(await ensureSsmInstall())) {
+      throw "Please try again after installing the required AWS utilities";
+    }
+
+    const { config } = await getAwsConfig(authn, request.accountId);
+    if (!config.login?.type || config.login?.type === "iam") {
+      throw "This account is not configured for SSH access via the P0 CLI";
+    }
+
+    return config.login?.type === "idc"
+      ? await assumeRoleWithIdc(request as AwsSshIdcRequest)
+      : config.login?.type === "federated"
+        ? await assumeRoleWithOktaSaml(authn, request as AwsSshRoleRequest)
+        : throwAssertNever(config.login);
+  },
+  proxyCommand: (request) => {
+    return [
+      "aws",
+      "ssm",
+      "start-session",
+      "--region",
+      request.region,
+      "--target",
+      "%h",
+      "--document-name",
+      START_SSH_SESSION_DOCUMENT_NAME,
+      "--parameters",
+      '"portNumber=%p"',
+    ];
+  },
+  reproCommands: (request) => {
+    // TODO: Modify commands to add the ability to get permission set commands
```

Comment:
I'm confused by this TODO. What's the thing we need to do?

Reply:
changed!